internal/http/compress.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+const (
+	// CompressorMinimumSize is the required size in bytes that a body exceeds before it is compressed
+	CompressorMinimumSize = 150
```

Comment:
Played around with this to determine a reasonable size for JSON to be gzip'd. There is a test JSON is 153 bytes to validate compression does happen.